sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/TableCatalogSuite.scala

```diff
@@ -813,9 +813,10 @@ class TableCatalogSuite extends SparkFunSuite {
 
     assert(catalog.namespaceExists(testNs) === false)
 
-    val ret = catalog.dropNamespace(testNs)
-
-    assert(ret === false)
+    val errMsg = intercept[NoSuchNamespaceException] {
+      catalog.dropNamespace(testNs)
```

Comment:
the previous behavior is better, that we return false if the namespace doesn't exist. Can we update `InMemoryTableCatalog.dropNamespace`?

Reply:
but the doc says about `NoSuchNamespaceException`, see https://github.com/apache/spark/blob/1aa184763aa49d70907669b2d8af5a713ee0d7fa/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsNamespaces.java#L146

Reply:
I think that from consistency point of view we should either:
- return empty array from `listTables()` and false from `dropNamespace()` for not existing namespace, or
- throw `NoSuchNamespaceException` it the provided namespace doesn't exist.

otherwise the implementation looks inconsistent.

Reply:
BTW, V1 `dropDatabase()` has parameter `ignoreIfNotExists` - it throws an exception if db doesn't exist

Reply:
I gonna to revert previous behavior of `dropNamespace` since it is out of PR's scope